skfda/preprocessing/dim_reduction/feature_extraction/_per_class_feature_transformer.py

```diff
@@ -0,0 +1,162 @@
+"""Feature extraction transformers for dimensionality reduction."""
+from __future__ import annotations
+
+from typing import TypeVar
+
+import numpy as np
+from sklearn.base import TransformerMixin
+from sklearn.utils.validation import check_is_fitted as sklearn_check_is_fitted
+
+from ...._utils import _fit_feature_transformer
+from ....representation.grid import FData
+
+T = TypeVar("T", bound=FData)
+
+
+class PerClassFeatureTransformer(TransformerMixin):
+    r"""Per class feature transformer for functional data.
+
+    This class takes a transformer and performs the following map:
+
+    .. math::
+        \mathcal{X} &\rightarrow \mathbb{R}^G \\
+        x &\rightarrow \textbf{t} = (T_1(x), T_2(x),...,T_k(x))
+
+    Where :math:`T_i(x)` is the transformation  :math:`x` with respect to
+    the data in the :math:`i`-th group.
+
+    Note that :math:`\mathcal{X}` is possibly multivariate, that is,
+    :math:`\mathcal{X} = \mathcal{X}_1 \times ... \times \mathcal{X}_p`.
+
+    Parameters:
+        transformer:
+            The transformer that we want to apply to the given data.
+            It should use target data while fitting.
+            This is checked by looking at the 'stateless' and 'requires_y' tags
+    Examples:
+        Firstly, we will import and split the Berkeley Growth Study dataset
+
+        >>> from skfda.datasets import fetch_growth
+        >>> from sklearn.model_selection import train_test_split
+        >>> X, y = fetch_growth(return_X_y=True, as_frame=True)
+        >>> X = X.iloc[:, 0].values
+        >>> y = y.values.codes
+        >>> X_train, X_test, y_train, y_test = train_test_split(
+        ...    X, y, test_size=0.25, stratify=y, random_state=0)
+
+        >>> from skfda.preprocessing.dim_reduction.feature_extraction
+        ... import PerClassFeatureTransformer
+
+        Then we will need to select a fda transformer, and so we will
+        use RecursiveMaximaHunting
+
+        >>> from skfda.preprocessing.dim_reduction.variable_selection
+        ... import RecursiveMaximaHunting
+
+        Finally we need to fit the data and transform it
+
+        >>> t.fit(X_train, y_train)
+        >>> x_transformed = t.transform(X_test)
+
+        x_transformed will be a vector with the transformed data
+    """
+
+    def __init__(
+        self,
+        transformer: TransformerMixin,
+    ) -> None:
+        self.transformer = transformer
+        self._validate_transformer()
+
+    def _validate_transformer(
+        self,
+    ) -> None:
+        """
+        Check that the transformer passed is\
+        scikit-learn-like and that uses target data in fit.
+
+        Args:
+            None
+
+        Returns:
+            None
+        """
+        if not (hasattr(self.transformer, "fit")
+                and hasattr(self.transformer, "fit_transform")
+                and hasattr(self.transformer, "transform")
+                ):
+
+            raise TypeError(
```

Comment:
It would be good to make this work also with [transductive](https://scikit-learn.org/stable/glossary.html#term-transductive) transformers (which only have `fit_transform`).

Reply:
I have to take my words back. Thinking more about it, it does not make sense to accept transductive transformers, as one always need to fit one class and transform the whole training set.